python/pyarrow/_parquet.pyx

```diff
@@ -1165,19 +1223,61 @@ cdef class ParquetReader(_Weakrefable):
         return pyarrow_wrap_chunked_array(out)
 
 
+cdef class LowLevelEncryptionProperties(_Weakrefable):
+    """Wrapper for FileEncryptionProperties.
+
+    Not intended for direct use by Python users, currently, just wrapped to
+    enable testing of low-level decryption.
+    """
+    cdef:
+        shared_ptr[FileEncryptionProperties] encryption_properties
+
+    def __cinit__(self, footer_key, column_keys=None, plaintext_footer=False,
+                  footer_key_metadata=None, column_keys_metadata=None):
+        cdef FileEncryptionProperties.Builder* builder
+        cdef c_string column
+        cdef c_map[c_string, shared_ptr[
+            ColumnEncryptionProperties]] c_column_keys
+
+        builder = new FileEncryptionProperties.Builder(footer_key)
```

Comment:
Is there a possibility of a memory leak here? Did you consider using a unique_ptr here instead?